libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/AudioCapabilities.java

```diff
@@ -102,25 +100,36 @@ public static AudioCapabilities getCapabilities(Context context) {
     if (Util.SDK_INT >= 23 && Api23.isBluetoothConnected(context)) {
       return DEFAULT_AUDIO_CAPABILITIES;
     }
+    int[] supportedEncodings = {};
```

Comment:
I have feeling that if we initialize this as a ImmutableSet.Builder, then add the encodings each time by:

```
// Below are all pseudo code

ImmutableSet.Builder supportedEncodings = new ImmutableSet.Builder<>();

supportedEncodings.addAll(Ints.asList(EXTERNAL_SURROUND_SOUND_ENCODINGS));
// Add other encodings

return AudioCapabilities(Ints.toArray(supportedEncodings.build()), DEFAULT_MAX_CHANNEL_COUNT);
```

In this way you don't need to add the de-duplication method. How do you think?

Reply:
One of the  duplications comes from the pcm 16 bit encoding that exists in extra_surround_sound_encoding and the directplaybacksupported encodings. Will the above pseudo code solve this case?

Reply:
We will add all the encodings (whether it is from `extra_surround_sound_encoding` or `directplaybacksupported`) to the `ImmutableSet.Builder` via `addAll` method right? And the [javadoc](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableSet.Builder.html#addAll(java.lang.Iterable)) of the method has such explanation:

```
Adds each element of elements to the ImmutableSet, ignoring duplicate elements (only the first duplicate element is added).
```
So as long as we add everything into the `ImmutableSet.Builder` and the `ImmutableSet.Builder.build()` produces a set (every element is unique), then this should solve the duplication?

Reply:
@tianyif I see. Will try that tomorrow.

Reply:
@tianyif  Done. Please see bc197dbd14499569708b0822bc1c9c016c6eabf3. Thanks